ops/model.py

```diff
@@ -388,6 +416,42 @@ def planned_units(self) -> int:
     def __repr__(self):
         return '<{}.{} {}>'.format(type(self).__module__, type(self).__name__, self.name)
 
+    def add_secret(self, content: Dict[str, str], *,
+                   label: Optional[str] = None,
+                   description: Optional[str] = None,
+                   expire: Optional[Union[datetime.datetime, datetime.timedelta]] = None,
+                   rotate: Optional['SecretRotate'] = None) -> 'Secret':
+        """Create a :class:`Secret` owned by this application.
+
+        Args:
+            content: A key-value mapping containing the payload of the secret,
+                for example :code:`{"password": "foo123"}`.
+            label: Private label to assign to this secret, which can later be
+                used for lookup.
+            description: Description of the secret's purpose.
+            expire: Time in the future (or timedelta from now) at which the
+                secret is due to expire. When that time elapses, Juju will
+                notify the charm by sending a SecretExpired event. None (the
+                default) means the secret will never expire.
+            rotate: Rotation policy/time. Every time this elapses, Juju will
+                notify the charm by sending a SecretRotate event. None (the
+                default) means to use the Juju default, which is never expire.
+        """
+        Secret._validate_content(content)
+        if isinstance(expire, datetime.timedelta):
+            expire = datetime.datetime.now() + expire
+        elif not isinstance(expire, datetime.datetime):
```

Comment:
if expire is None (the default value), resulting in exception

  File "/var/lib/juju/agents/unit-keystone-0/charm/./src/charm.py", line 302, in _retrieve_or_set_secret
    credentials_secret = self.model.app.add_secret(
  File "/var/lib/juju/agents/unit-keystone-0/charm/venv/ops/model.py", line 444, in add_secret
    raise TypeError('Expiration time must be a datetime or timedelta from now, not '
TypeError: Expiration time must be a datetime or timedelta from now, not NoneType